### PR TITLE
Add pool with max_backlog option (fixes #509)

### LINKED ISSUE #509: SQLA Queue with max backlog
As stated #186, it could be nice to have a max_backlog option for the connection pool. Since it is not native in SQLAlchemy it was not part of #231 

This was implemented in Sync a while ago:
https://github.com/mozilla-services/server-syncstorage/commit/f93ca85a985364f123ad79cbdc2fcc29c5656586

We could: 
- discuss with SQLA to see it would be something relevant to contribute
- import and test the existing code otherwise 
- set it as default connection queue when creating the engine

### PR DESCRIPTION
**Pull request by leplatrem**
This code from imported from Mozilla Services Sync.
See mozilla-services/server-syncstorage@f93ca85

*Note: Cliquet is licensed under Apache and Sync is under MPL*

@Natim r?
@tarekziade, @rfk  comments?

### CODE REVIEW

File: cliquet/cache/postgresql/__init__.py [hunk 1]
```diff
@@ -53,7 +59,7 @@ class Cache(CacheBase):
         of connections used in a multi-process deployment.
 
     :noindex:
-    """
+    """  # NOQA
```
Comment (Natim) [inline cliquet/cache/postgresql/__init__.py]: Why would you need ``# NOQA`` shouldn't we let an empty line before instead?

**Comment by tarekziade** [comment]
Back to the original reason why this was done : https://bugzilla.mozilla.org/show_bug.cgi?id=750964

I am curious if we've reproduced this issue in our load tests ?

File: cliquet/cache/postgresql/__init__.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/cache/postgresql/__init__.py]: Because of the 80 chars limit of the default ``cliquet.cache_poolclass`` line :]

**Comment by leplatrem** [comment]
> I am curious if we've reproduced this issue in our load tests ?

Not that I am aware of :)

**Comment by leplatrem** [comment]
@Natim bump r?

**Comment by Natim** [comment]
r+

**Pull request merged**